geaflow/geaflow-dsl/geaflow-dsl-plan/src/main/java/com/antgroup/geaflow/dsl/udf/table/string/Instr.java

```diff
@@ -44,4 +45,29 @@ public Long eval(String str, String target, Long from, Long nth) {
         }
         return (long) fromIndex;
     }
+
+    public Long eval(BinaryString str, BinaryString target) {
+        return eval(str, target, 1L, 1L);
+    }
+
+    public Long eval(BinaryString str, BinaryString target, Long from) {
+        return eval(str, target, from, 1L);
+    }
+
+    public Long eval(BinaryString str, BinaryString target, Long from, Long nth) {
+        if (str == null || target == null || from == null || nth == null) {
+            return null;
+        }
+        if (nth <= 0) {
+            return null;
+        }
+        int fromIndex = from.intValue() - 1;
+        if (fromIndex < 0) {
```

Comment:
Why do we need to return null here?

Reply:
There may be multiple interpretations here. My understanding is that fromIndex < 0 can be regarded as starting the search from the beginning of the string. However, in Aliyun's ODPS, a negative start_position is considered to indicate that the starting position is counted backwards from the end of the string. Refer to https://help.aliyun.com/document_detail/455546.html?spm=a2c4g.48973.0.i1 for more information.